pkg/ali/components/CruACK.vue

```diff
@@ -0,0 +1,656 @@
+  runtime:        'containerd',
+  runtimeVersion: '1.6.38',
+  vswitchIds:     [],
+
+  _isNewOrUnprovisioned: true,
+};
+
+export default defineComponent({
+  name:       'CruACK',
+  emits:      ['validationChanged', 'error'],
+  components: {
+    SelectCredential,
+    CruResource,
+    LabeledSelect,
+    LabeledInput,
+    ClusterMembershipEditor,
+    Labels,
+    Accordion,
+    Banner,
+    Loading,
+    Import,
+    Networking,
+    NodePool,
+    Tabbed,
+    Tab,
+    AgentConfiguration
+  },
+
+  mixins: [CreateEditView, FormValidation],
+
+  props: {
+    mode: {
+      type:    String,
+      default: _CREATE
+    },
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    }
+  },
+  data() {
+    return {
+      normanCluster:          { name: '' },
+      nodePools:              [],
+      membershipUpdate:       {},
+      originalVersion:        '',
+      locationOptions:        [],
+      allVersions:            [],
+      allImages:              {},
+      allInstanceTypes:       {},
+      loadingLocations:       false,
+      loadingVersions:        false,
+      loadingInstanceTypes:    false,
+      loadingImages:          false,
+      fvFormRuleSets:         [],
+      configUnreportedErrors: [],
+      configIsValid:          true,
+      zones:                  new Set()
+    };
+  },
+
+  created() {
+    const registerAfterHook = this.registerAfterHook ;
+
+    registerAfterHook(this.saveRoleBindings, 'save-role-bindings');
+  },
+
+  async fetch() {
+    const store = this.$store;
+
+    if (this.value.id) {
+      const liveNormanCluster = await this.value.findNormanCluster();
+
+      this.normanCluster = await store.dispatch(`rancher/clone`, { resource: liveNormanCluster });
+    } else {
+      const base = !this.isImport ? defaultCluster : importedDefaultCluster ;
+
+      this.normanCluster = await store.dispatch('rancher/create', { type: NORMAN.CLUSTER, ...base }, { root: true });
+    }
+    if (this.isImport) {
+      this.normanCluster.aliConfig = cloneDeep(importedDefaultAckConfig);
+    } else {
+      if (!this.normanCluster.aliConfig) {
+        this.normanCluster.aliConfig = { ...defaultAckConfig };
+      }
+      if (this.mode === _CREATE && (!this.normanCluster.aliConfig.nodePools || this.normanCluster.aliConfig.nodePools.length === 0)) {
+        const pool = cloneDeep(DEFAULT_NODE_GROUP_CONFIG);
+
+        pool._id = randomStr();
+        this.normanCluster.aliConfig.nodePools = [pool];
+      } else {
+        this.normanCluster.aliConfig.nodePools.forEach((pool) => {
+          pool['_id'] = pool.nodePoolId || randomStr();
+          pool['_isNewOrUnprovisioned'] = this.isNewOrUnprovisioned;
+          pool['_validation'] = {};
+        });
+      }
+      this.nodePools = this.normanCluster.aliConfig.nodePools;
+    }
+  },
+  watch: {
+    'config.alibabaCredentialSecret'(neu) {
+      if (neu) {
+        this.getLocations();
```

Comment:
I think all of the fields that depend on fetching data from alibaba need to re-compute when a new credential is selected not just location